### PR TITLE
[WIP] Generate payload types

### LINKED ISSUE #43: media engine: generate payload types
When creating a codec, dynamically generate a payload type in the range 96-127 if one wasn't provided. The generation is described in [RFC3551](https://tools.ietf.org/html/rfc3551#section-3)

### PR DESCRIPTION
**Pull request by porjo**
Resolves #43

### CODE REVIEW

File: mediaengine.go [hunk 1]
```diff
@@ -14,51 +14,52 @@ func RegisterCodec(codec *RTCRtpCodec) {
...
 
-func (m *MediaEngine) getCodec(payloadType uint8) (*RTCRtpCodec, error) {
-	for _, codec := range m.codecs {
-		if codec.PayloadType == payloadType {
-			return codec, nil
+	var payloadType uint8
+
+	// check for existing payload type
+	for _, c := range m.codecs {
+		if c.Name == codec.Name {
```
Comment (porjo) [inline mediaengine.go]: Is it sufficient to identify the codec by name only?

Comment (backkem) [reply mediaengine.go]: The payload type defines a specific codec, including all its properties. Looking up a codec based on its properties can be done like in ``getCodecSDP``.

**Comment by porjo** [comment]
> Those applications which need to define more than 32 dynamic payload types

That seems like an extreme use case to need more than 32 payload types per peer connection. Do we need to cater for that?

**Comment by Sean-Der** [comment]
Hey @porjo really sorry this took so long, when I get home from work will get you the feedback needed to get merged.

thanks!

**Comment by Sean-Der** [comment]
Hey @porjo I am really sorry this got ignored. That was really bad on our part, I passed this off and didn't realize you were waiting on feedback from us.

Hopefully this didn't push you away from using Pion! I am going to close this for now, I am going to go back and fix how painful it is to use the Media APIs in the next release.

**Pull request closed without merging**